handler/src/main/java/io/netty/handler/ssl/util/JavaKeyStore.java

```diff
@@ -0,0 +1,152 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl.util;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+/**
+ * Allows to easily extract a {@link PrivateKey} and the {@link X509Certificate} chain out of a {@link KeyStore}.
+ */
+public final class JavaKeyStore {
+
+    private final X509Certificate[] certChain;
+    private final PrivateKey key;
+
+    /**
+     * Creates a new {@link JavaKeyStore} instance.
+     *
+     * @param jks the JKS file from which load the {@link KeyStore}.
+     * @param alias the alias or {@code null} if we should try to auto-detect.
+     * @param password the password or {@code null} if non is needed to load the {@link KeyStore}.
+     * @throws KeyStoreException
+     * @throws UnrecoverableKeyException
+     * @throws IOException
+     * @throws NoSuchAlgorithmException
+     * @throws CertificateException
+     */
+    public JavaKeyStore(File jks, String alias, String password)
+            throws KeyStoreException, UnrecoverableKeyException, IOException,
+            NoSuchAlgorithmException, CertificateException {
+        this(keyStore(checkNotNull(jks, "jks"), password), alias, password);
+    }
+
+    /**
+     * Creates a new {@link JavaKeyStore} instance.
+     *
+     * @param ks the {@link KeyStore}.
+     * @param alias the alias or {@code null} if we should try to auto-detect.
+     * @param password the password or {@code null} if non is needed to load the {@link KeyStore}.
+     * @throws KeyStoreException
+     * @throws UnrecoverableKeyException
+     * @throws NoSuchAlgorithmException
+     */
+    public JavaKeyStore(KeyStore ks, String alias, String password)
+            throws KeyStoreException, UnrecoverableKeyException, NoSuchAlgorithmException {
+        alias = alias(checkNotNull(ks, "ks"), alias);
+        certChain = certificateChain(ks, alias);
+        key = privateKey(ks, alias, password);
+    }
+
+    /**
+     * Returns the {@link PrivateKey}.
+     */
+    public PrivateKey key() {
+        return key;
+    }
+
+    /**
+     * Return the {@link X509Certificate} chain.
+     */
+    public X509Certificate[] certificateChain() {
```

Comment:
I wonder if it's a good idea to return a `List<X509Certificate>` instead of an array.